### PR TITLE
Patch RNG for better memory utilization in dropout layers

### PR DESCRIPTION
**Pull request by yeounoh**
The patch has landed in OpenXLA https://github.com/openxla/xla/pull/6015, we can enable this now, to reduce memory pressure in `nn.Dropout` layer.

DO_NOT_MERGE until the next pin update.

### CODE REVIEW

**Review by JackCaoG — APPROVED** [review]

**Comment by JackCaoG** [comment]
do we want to add a test to check that int8 is used for rng? I think we can check the HLO generated,

**Comment by yeounoh** [comment]
> do we want to add a test to check that int8 is used for rng? I think we can check the HLO generated,

Hey Jack, yea good idea, will add before merging.

**Comment by yeounoh** [comment]
> > do we want to add a test to check that int8 is used for rng? I think we can check the HLO generated,
> 
> Hey Jack, yea good idea, will add before merging.

Verified locally with the new unit test -- cc @JackCaoG
